org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -141,7 +143,7 @@ public IStatus runInWorkspace(IProgressMonitor monitor) {
 
 	private void deleteInvalidProjects(Collection<IPath> rootPaths, IProgressMonitor monitor) {
 		for (IProject project : getWorkspaceRoot().getProjects()) {
-			if (project.exists() && ResourceUtils.isContainedIn(project.getLocation(), rootPaths)) {
+			if (project.exists() && (ResourceUtils.isContainedIn(project.getLocation(), rootPaths) || ProjectUtils.isGradleProject(project))) {
```

Comment:
are we deleting all existing gradle projects?!

Reply:
We mustn't delete any valid gradle project. If you open sagan-common, buildship will open sagan, sagan-client...
Those projects can be anywhere in the file system. See https://github.com/spring-io/sagan/blob/master/settings.gradle